### ISSUE TITLE
BigQuery: Add support for regional resources

### ISSUE THREAD

**Issue by fhoffa** [getredash/redash#3260]
BigQuery now offers customers multiple regions for data querying and storage, for example:

  | Region Name | Region Description
-- | -- | --
  | northamerica-northeast1 | Montréal
  | us-east4 | Northern Virginia
  | europe-north1 | Finland
  | europe-west2 | London
  | asia-south1 | Mumbai
  | asia-east1 | Taiwan
  | asia-northeast1 | Tokyo
  | asia-southeast1 | Singapore
  | australia-southeast1 | Sydney

To support Regional BigQuery the BQ team added a location string to the BQ API, which can be a breaking change. BigQuery performs auto-detect for jobs.insert.  However, getqueryresults(), jobs.get(), and jobs.cancel need a location to direct the command to the appropriate region.

Related docs: https://cloud.google.com/bigquery/docs/locations

----

tl;dr: The BigQuery jobs resource now contains a location.  It is optional during jobs.insert and jobs.query, but required for jobs.get, jobs.cancel, and jobs.getQueryResults.

**Comment by arikfr** [comment]
Thanks, @fhoffa ! This is already supported since v5 :)
